### DISCUSSION TITLE
Let's Encrypt SSL context leads to "CERTIFICATE_VERIFY_FAILED"

### DISCUSSION THREAD

**Discussion by flowli** [mjs/imapclient#440] [Q&A]
Dear IMAPClient community,

thank you for this wonderful project.

I am trying to connect to an IMAP server with Let's Encrypt certificate.

My observations so far:

1. without an ssl_context given to IMAPClient during instanciation it leads to an CERTIFICATE_VERIFY_FAILED error.
2. the following example script provides an ssl_context to IMAPClient, yet fails with CERTIFICATE_VERIFY_FAILED, too:
````
#!/usr/bin/env python

from imapclient import IMAPClient
import ssl

config = {
    'imap_cert_file': '../ssl/fullchain.pem',
    'imap_key_file': '../ssl/privkey.pem',
    'imap_host': 'mail.example.com'
}

ssl_context = ssl.create_default_context()
ssl_context.load_cert_chain(config['imap_cert_file'], config['imap_key_file'])

IMAPClient(config['imap_host'], ssl_context=ssl_context)
````

My question: what am I doing wrong, in order to connect successfully with my IMAP server?

Thank you for any pointers!

**Comment by mjs** [comment]
Just to confirm, the IMAP server is using certificates issued by Let's Encrypt?

**Reply by flowli** [reply]
Yes, it does.

**Comment by flowli** [comment]
I switched to a ☁️ provider which does not require explicitly defining a custom certificate. But would still be interested how to solve this. Any ideas?
